src/Orleans/Runtime/IGrainRuntime.cs

```diff
@@ -41,5 +42,7 @@ public interface IGrainRuntime
         void DeactivateOnIdle(Grain grain);
 
         void DelayDeactivation(Grain grain, TimeSpan timeSpan);
+
+        IStorage<TGrainState> GetStateStorageBridge<TGrainState>(Grain<TGrainState> grain) where TGrainState : new();
```

Comment:
Storage providers and this behavior are not limited to Grain<T>.
Suggest:
IStorage<TGrainState> GetStateStorageBridge<TGrainState>(Grain grain)

Reply:
Not sure to follow you?